graph/src/data/schema.rs

```diff
@@ -73,6 +73,111 @@ pub enum SchemaValidationError {
         _0, _1
     )]
     ImportedTypeUndefined(String, String), // (type_name, schema)
+    #[fail(display = "Fulltext directive name undefined")]
+    FulltextNameUndefined,
+    #[fail(display = "Fulltext directive name overlaps with type: {}", _0)]
+    FulltextNameOverlap(String),
+    #[fail(display = "Fulltext language is undefined")]
+    FulltextLanguageUndefined,
+    #[fail(display = "Fulltext language is invalid: {}", _0)]
+    FulltextLanguageInvalid(String),
+    #[fail(display = "Fulltext algorithm is undefined")]
+    FulltextAlgorithmUndefined,
+    #[fail(display = "Fulltext algorithm is invalid: {}", _0)]
+    FulltextAlgorithmInvalid(String),
+    #[fail(display = "Fulltext include undefined")]
+    FulltextIncludeUndefined,
+    #[fail(display = "Fulltext include invalid")]
+    FulltextIncludeInvalid,
+}
+
+enum FulltextLanguage {
+    Simple,
+    Danish,
+    Dutch,
+    English,
+    Finnish,
+    Fresh,
```

Comment:
French?

Reply:
Maybe we should support `Fresh` as well?